dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/CodeBlock.cs

```diff
@@ -165,12 +167,29 @@ private KernelArguments EnrichFunctionArguments(KernelArguments arguments)
             this.Logger.LogTrace("Passing variable/value: `{Content}`", firstArg.Content);
         }
 
+        string? positionalParameterName = null;
+        object? positionalInputValue = null;
         var namedArgsStartIndex = 1;
+
+        // Get the function metadata
+        var functionMetadata = kernel.Plugins.GetFunction(fBlock.PluginName, fBlock.FunctionName).Metadata;
+
+        // Check if the function has parameters to be set
+        if (functionMetadata.Parameters.Count == 0)
+        {
+            throw new ArgumentException($"Function {fBlock.PluginName}.{fBlock.FunctionName} has no parameters.");
+        }
```

Comment:
It's valid for a plugin function to have no parameters so this message is confusing. What is the reason that it's valid to throw an exception here?

Reply:
This only happens if you provide parameters to a parameterless function in your template.

Function
```csharp
void FunctionA()
```

Template
```
{{FunctionA 'abc'}}
```

This will throw.